core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -522,10 +522,11 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
         }
     }
 
-    public Engine.Index prepareIndexOnReplica(SourceToParse source, long seqNo, long version, VersionType versionType, long autoGeneratedIdTimestamp,
-                                              boolean isRetry) {
+    public Engine.Index prepareIndexOnReplica(SourceToParse source, long seqNo, long primaryTerm, long version, VersionType versionType,
```

Comment:
> Here it means the term of the primary that was sending this operation, not necessarily the primary term of the log entry that is being replicated

Actually it's the term of the op in the log and not the authority of the primary that sends this op (that one gause into the permit method). I guess this proves your point about being confusing. Any suggestion?